### PR TITLE
added pagination for filling out a survey

### PR DESCRIPTION
**Pull request by Shashaank5**

### CODE REVIEW

File: src/main/java/com/based/basedsurvey/controller/FillSurveyController.java [hunk 1]
```diff
@@ -54,29 +60,87 @@ public String getAnswer(@PathVariable Long surveyID, Model model) {
...
+
+        // if survey is still finished
+        if (s.getStatus() == Survey.SurveyStatuses.FINISHED) {
+            model.addAttribute("issue", "Survey is finished");
+            return "SurveyFillIssue";
+        }
+
+        // if survey has no questions
+        if (s.getQuestions().size() == 0) {
+            model.addAttribute("issue", "Survey has no questions");
```
Comment (DavidHaighton) [inline src/main/java/com/based/basedsurvey/controller/FillSurveyController.java:113]: based defensive programming

File: src/main/java/com/based/basedsurvey/controller/FillSurveyController.java [hunk 2]
```diff
@@ -54,29 +60,87 @@ public String getAnswer(@PathVariable Long surveyID, Model model) {
...
+                // case 2, the question is an open answer question
+            } else if (q instanceof OpenAnswerQuestion) {
+                formInputs.append(getOpenAnswerInput((OpenAnswerQuestion) q));
+
+                // case 3, the question is a range question
+            } else if (q instanceof RangeQuestion) {
+                formInputs.append(getRangeInput((RangeQuestion) q));
+            }
+        }
+        formInputs.append(getPaginateHtml(surveyID, page + 1));
```
Comment (DavidHaighton) [inline src/main/java/com/based/basedsurvey/controller/FillSurveyController.java:140]: does this work even if no pages are left?

Comment (Shashaank5) [reply src/main/java/com/based/basedsurvey/controller/FillSurveyController.java:91]: yes there is a check earlier if paginated query is empty

File: src/test/java/com/based/basedsurvey/controller/FillSurveyControllerTest.java [hunk 3]
```diff
@@ -139,16 +139,6 @@ public void testFillSurvey() throws Exception {
         String param2 = "values" + oq.getId();
         String param3 = "values" + rq.getId();
 
-        // get the page to fill out the survey and expect it contains all the question prompts
```
Comment (Shashaank5) [inline src/test/java/com/based/basedsurvey/controller/FillSurveyControllerTest.java:142]: Test is no longer relevant, it's doesn't make sense to test some changes due to the additions in HTMX

**Review by DavidHaighton — APPROVED** [review]
nice using pagination

File: src/test/java/com/based/basedsurvey/controller/FillSurveyControllerTest.java [hunk 4]
```diff
@@ -139,16 +139,6 @@ public void testFillSurvey() throws Exception {
...
-        this.mockMvc.perform(get("/survey/1/answer")).andDo(print())
-                .andExpect(content().string(containsString("Is this project super based or ultra based?:")))
-                .andExpect(content().string(containsString("How based is this project?")))
-                .andExpect(content().string(containsString("Rate how based this project is")))
-                .andExpect(content().string(containsString("additional notes: hint, it's ultra based")))
```
Comment (DavidHaighton) [inline src/test/java/com/based/basedsurvey/controller/FillSurveyControllerTest.java:147]: issue with unit test is that its no longer relevant and htmx is making it hard to test

**Review by DavidHaighton — APPROVED** [review]

**Pull request merged**